autoPlay.js

```diff
@@ -1,458 +1,510 @@
+    spawnerExists: true,
+    bossMinHealthPercent: null,
+    spawnerMinHealthPercent: 0.95
+};
+
+if (thingTimer) {
+    window.clearInterval(thingTimer);
 }
 
 function firstRun() {
-	// disable particle effects - this drastically reduces the game's memory leak
-	if (window.g_Minigame !== undefined) {
-		window.g_Minigame.CurrentScene().DoClickEffect = function() {};
-		window.g_Minigame.CurrentScene().DoCritEffect = function( nDamage, x, y, additionalText ) {};
-		window.g_Minigame.CurrentScene().SpawnEmitter = function(emitter) {
-			emitter.emit = false;
-			return emitter;
-		}
-	}
-
-	// disable enemy flinching animation when they get hit
-	if (window.CEnemy !== undefined) {
-		window.CEnemy.prototype.TakeDamage = function() {};
-		window.CEnemySpawner.prototype.TakeDamage = function() {};
-		window.CEnemyBoss.prototype.TakeDamage = function() {};
-	}
-
-	if (thingTimer !== undefined) {
-		window.clearTimeout(thingTimer);
-	}
+    // disable particle effects - this drastically reduces the game's memory leak
+    if (g_Minigame !== undefined) {
+        //disableDamageText();
+        g_Minigame.CurrentScene().SpawnEmitter = function (emitter) {
+            emitter.emit = false;
+            return emitter;
+        }
+    }
+
+    // disable enemy flinching animation when they get hit
+    //disableFlinchingAnimation();
+    // too many confused users think that the script breaks clicking. The flinching animation uses few resources and is a good enough indicator.
 }
 
 function doTheThing() {
-	if (!isAlreadyRunning){
-		isAlreadyRunning = true;
-
-		goToLaneWithBestTarget();
-		useGoodLuckCharmIfRelevant();
-		useMedicsIfRelevant();
-		useMoraleBoosterIfRelevant();
-		useClusterBombIfRelevant();
-		useNapalmIfRelevant();
-		attemptRespawn();
-
-		isAlreadyRunning = false;
-	}
+    if (!isAlreadyRunning) {
+        isAlreadyRunning = true;
+
+        goToLaneWithBestTarget();
+        useOffensiveAbilityIfAvailable(ABILITIES.GOOD_LUCK);
+        useMedicsIfRelevant();
+        useOffensiveAbilityIfAvailable(ABILITIES.MORALE_BOOSTER);
+        useOffensiveAbilityIfAvailable(ABILITIES.CLUSTER_BOMB);
+        useOffensiveAbilityIfAvailable(ABILITIES.NAPALM);
+        useOffensiveAbilityIfAvailable(ABILITIES.NUKE);
+        useItemIfAvailable(ITEMS.CRIPPLE_SPAWNER);
+        useItemIfAvailable(ITEMS.GOLD_RAIN);
+        attemptRespawn();
+
+        isAlreadyRunning = false;
+    }
 }
 
 function goToLaneWithBestTarget() {
-	// We can overlook spawners if all spawners are 40% hp or higher and a creep is under 10% hp
-	var spawnerOKThreshold = 0.4;
-	var creepSnagThreshold = 0.1;
-	
-	var targetFound = false;
-	var lowHP = 0;
-	var lowLane = 0;
-	var lowTarget = 0;
-	var lowPercentageHP = 0;
-	
-	// determine which lane and enemy is the optimal target
-	var enemyTypePriority = [
-		ENEMY_TYPE.TREASURE, 
-		ENEMY_TYPE.BOSS, 
+    // We can overlook spawners if all spawners are 40% hp or higher and a creep is under 10% hp
```

Comment:
Please revert the whitespace changes. The code is formatted with tabs, and you've switched a bunch of it to spaces.